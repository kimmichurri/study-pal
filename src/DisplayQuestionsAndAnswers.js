import React from 'react';
import StartOverNotice from './StartOverNotice';

export default function DisplayQuestionsAndAnswers(props) {
    
        if (props.currentIndex === props.allFlashcards.length) {
            return (
            <StartOverNotice />
            )
        }

        if (props.correctAnswer) {
            return (
                <section>
                    <p className="question-text">That's right!</p>
                    <button onClick={props.showNextFlashcard} className="answer-text correct-answer">Crushed it! Keep Going!</button>
                </section>
            )
         } else if (props.correctAnswer === false) {
            return (
                <section>
                    <p className="question-text">Nice try, but incorrect. Let's add this question to your Study Bank for later!</p>
                    <button onClick={props.showNextFlashcard} className="answer-text incorrect-answer">Keep Trying!</button>
                </section>
             )
         } 

       return (
           <section>
               <p className="question-text">{props.flashcard.question}</p>
               <section className="answer-choice-buttons-container">
                   <button onClick={props.validateAnswer} className="answer-text">{props.flashcard.answerChoices[0]}</button>
                   <button onClick={props.validateAnswer} className="answer-text">{props.flashcard.answerChoices[1]}</button>
                   <button onClick={props.validateAnswer} className="answer-text">{props.flashcard.answerChoices[2]}</button>
               </section>
           </section>
       )
}