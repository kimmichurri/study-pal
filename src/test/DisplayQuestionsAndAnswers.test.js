import React from 'react';
import DisplayQuestionsAndAnswers from '../DisplayQuestionsAndAnswers';
import { shallow } from 'enzyme';

//this is what gets passed to this component:
{/* <DisplayQuestionsAndAnswers 
                    flashcard={this.props.flashcards[this.props.currentIndex]} 
                    validateAnswer={this.validateAnswer}
                    correctAnswer={this.state.correctAnswer}
                    showNextFlashcard={this.showNextFlashcard}
                    currentIndex={this.props.currentIndex}
                    allFlashcards={this.props.flashcards}
                />  */}

const mockAllFlashcards = [
    {
        "id": "a257e858-36c6-11e9-b210-d663bd873d93",
        "category": "String Methods",
        "question": "Which method allows you to remove whitespace from either end of a string?",
        "answerChoices": [".trim()", ".slim()", ".clip()"],
        "correctAnswer": ".trim()",
        "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim"
    }
]

const mockQuestion = mockAllFlashcards[0].question;

let mockIndex = 1;

let mockCorrectAnswer;

describe('DisplayQuestionsAndAnswers', () => {
    let wrapper;


    it('should match the snapshot if props.currentIndex === props.allFlashcards.length', () => {

        const wrapper = shallow( 
            <DisplayQuestionsAndAnswers 
                    flashcard={mockAllFlashcards[mockIndex]} 
                    currentIndex={mockIndex}
                    allFlashcards={mockAllFlashcards}
                /> 
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot if props.correct answer', () => {
        mockCorrectAnswer = true;
        mockIndex = 2;

        const wrapper = shallow(
            <DisplayQuestionsAndAnswers 
                    flashcard={mockAllFlashcards[mockIndex]} 
                    correctAnswer={mockCorrectAnswer}
                    currentIndex={mockIndex}
                    allFlashcards={mockAllFlashcards}
                />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot if props.correct answer === false', () => {
        mockCorrectAnswer = false;
        mockIndex = 2;

        const wrapper = shallow(
            <DisplayQuestionsAndAnswers 
                flashcard={mockAllFlashcards[mockIndex]} 
                correctAnswer={mockCorrectAnswer}
                currentIndex={mockIndex}
                allFlashcards={mockAllFlashcards}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should return a flashcard', () => {
        mockIndex = 0;
        wrapper = shallow(
            <DisplayQuestionsAndAnswers 
                flashcard={mockAllFlashcards[mockIndex]} 
                currentIndex={mockIndex}
                allFlashcards={mockAllFlashcards}
            />
        )
        expect(wrapper).toMatchSnapshot();
    });

});