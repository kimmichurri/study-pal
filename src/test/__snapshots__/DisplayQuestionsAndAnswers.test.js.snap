// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayQuestionsAndAnswers should match the snapshot if props.correct answer === false 1`] = `
<section>
  <p
    className="question-text"
  >
    Nice try, but incorrect. Let's add this question to your Study Bank for later!
  </p>
  <button
    className="answer-text"
  >
    Keep Trying!
  </button>
</section>
`;

exports[`DisplayQuestionsAndAnswers should match the snapshot if props.correct answer 1`] = `
<section>
  <p
    className="question-text"
  >
    That's right!
  </p>
  <button
    className="answer-text"
  >
    Crushed it! Keep Going!
  </button>
</section>
`;

exports[`DisplayQuestionsAndAnswers should match the snapshot if props.currentIndex === props.allFlashcards.length 1`] = `<StartOverNotice />`;

exports[`DisplayQuestionsAndAnswers should return a flashcard 1`] = `
<section>
  <p
    className="question-text"
  >
    Which method allows you to remove whitespace from either end of a string?
  </p>
  <section
    className="answer-choice-buttons-container"
  >
    <button
      className="answer-text"
    >
      .trim()
    </button>
    <button
      className="answer-text"
    >
      .slim()
    </button>
    <button
      className="answer-text"
    >
      .clip()
    </button>
  </section>
</section>
`;
